File: BackEnd/Domain/Aggregates/Comment.cs
using SharedKernel.Aggregate;

namespace Domain.Aggregates;

public class Comment : AggregateRoot<Guid>
{
    public Guid TicketId { get; set; }
    public Guid UserId { get; set; }
    public Ticket Ticket { get; set; }
    public string Content { get; set; }
    
    public User User { get; set; }

    public static Comment Create(Guid ticketId, string content, Guid userId)
    {
        return new Comment
        {
            Id = Guid.CreateVersion7(),
            TicketId = ticketId,
            Content = content,
            UserId = userId
        };
    }
    
    public void Update(string content)
    {
        Content = content;
    }
}

File: flutter/lib/domain/aggregates/comment/comment.dart
import 'package:dart_mappable/dart_mappable.dart';
import 'package:enterprise_management/shared_kernel/aggregate/aggregate_root.dart';

part 'comment.mapper.dart';
@MappableClass()
class Comment extends AggregateRoot<String> with CommentMappable {
  Comment({required super.id, required this.content, required this.userId, required this.ticketId});

  final String content;
  final String userId;
  final String ticketId;
}